third-party/realdds/doc/control.md

```diff
@@ -14,16 +14,73 @@ I.e., when a control is sent, some way to identify a reply to that specific cont
 
 - Replies will have the same `id` as the original control. This means controls and notifications must have separate identifiers.
 
-Additionally, when the server gets a control message, the sample will contain a GUID for its publisher (called the "publication handle"). This is the GUID of the writer used. The participant can be deduced from this and attached to a reply:
+Additionally, when the server gets a control message, the DDS sample will contain a "publication handle". This is the GUID of the writer (including the participant) to identify the control source. A unique sequence number for each control sample is available from the `sample-identity`.
 
-Additionally, a unique sequence number for each such sample is available from the `sample-identity`.
-
-- Replies will contain a `sample` field containing the GUID of the participant from which the control request originated, plus the sequence-number, e.g.: `"sample": "010f6ad8708c95e000000000.4"`
+- Replies will contain a `sample` field containing the GUID of the participant from which the control request originated, plus the sequence-number, e.g.: `"sample": ["010f6ad8708c95e000000000.303",4]`
 
 Replies must include a success/failure status:
 
 - Replies will include a `status` for anything other than `OK`
-- Replies will include an `explanation` is status is not `OK`
+- Replies will include an `explanation` if status is not `OK`
+
+Replies may also contain the original control request:
+
+- Optionally (depending on the control), a `control` field in the reply should contain the original control request (rather than copying many fields, for example)
+
+
+### Errors
+
+An error is a reply where the `status` is not OK:
+
+- the `explanation` is a human-readable string explaining the failure (like an exception's `what`)
+
+```JSON
+{
+    "id": "query-option",
+    "sample": ["010f9a5f64d95fd300000000.403", 1],
+    "control": {
+        "id": "query-option",
+        "option-name": "opt16"
+        },
+    "status": "error",
+    "explanation": "Test Device does not support option opt16"
+}
+```
+
+
+## Control Messages
+
+### `query-option` & `set-option`
+
+Returns or sets the current `value` of an option within a device.
+
+- `id` is `query-option` or `set-option`
+- `option-name` is the unique option name within its owner
+- `owner-name` is the name of the owner
+    - for a device option, this may be omitted
+    - for a stream option, this is the unique stream name within the device
+
+```JSON
+{
+    "id": "query-option",
+    "option-name": "opt-name"
+}
```

Comment:
Should also give an example of set options, it has more fields

Reply:
Just one more, `value`, which can be seen in the reply -- but added an explicit description.